File: DSA/SakshamDwivedi_590013147/Day10/D10P1.py
#Solution 1 : Less efficient
def compare(string1, string2):
    if(string1 == string2):
        print("Equal, ", end = " ")
    else:
        print("Not equal, ", end = " ")
    if(string1.lower() == string2.lower()):
        print("Equal ignoring case, ", end = " ")
    else:
        print("Not equal ignoring case, ", end = " ")
    if(string1 > string2):
        print(string2 + " comes before " + string1)
    elif(string2 > string1):
        print(string1 + " comes before " + string2)
    else:
        print("Both strings are identical!")


#Solution 2 : This one is more efficient as it doesn't perform all checks
def compare2(string1, string2):
    if(string1 == string2):
        print("Equal, Equal ignoring case, Both strings are identical")
    else:
        print("Not equal, ", end = " ")
        if(string1.lower() == string2.lower()):
            print("Equal ignoring case, ", end = " ")
        else:
            print("Not equal ignoring case, ", end = " ")
        if(string1 > string2):
            print(string2 + " comes before " + string1)
        else:
            print(string1 + " comes before " + string2)
        

#Fixed Input
compare("Hello", "hello")
compare("test", "test")

compare2("Hello", "hello")
compare2("test", "test")

#Custom Input
cstring1 = input("Enter string 1 : ")
cstring2 = input("Enter string 2 : ")
compare(cstring1, cstring2)
compare2(cstring1, cstring2)

File: DSA/SakshamDwivedi_590013147/Day10/D10P1.c
#include <stdio.h>
#include <string.h>

char string1[100];
char string2[100];
int size1;
int size2;
char lower1[100];
char lower2[100];


int main(){
    printf("Enter the string 1 (Less than 100 characters) : ");
    fgets(string1, 100, stdin);
    printf("Enter the string 2 (Less than 100 characters) : ");
    fgets(string2, 100, stdin);
    if(string1[strlen(string1) - 1] == '\n'){
        string1[strlen(string1) - 1] = '\0';
    }
    size1 = strlen(string1);

    if(string2[strlen(string2) - 1] == '\n'){
        string2[strlen(string2) - 1] = '\0';
    }
    size2 = strlen(string2);

    for(int i = 0; i < size1; i++){
        if(string1[i] >= 'A' && string1[i] <= 'Z'){
            lower1[i] = string1[i] + 32;
        }
        else{
            lower1[i] = string1[i];
        }
    }
    lower1[size1] = '\0';

    for(int i = 0; i < size2; i++){
        if(string2[i] >= 'A' && string2[i] <= 'Z'){
            lower2[i] = string2[i] + 32;
        }
        else{
            lower2[i] = string2[i];
        }
    }
    lower2[size2] = '\0';

    int scmp = strcmp(string1, string2);

    if(scmp == 0){
        printf("Equal, Equal ignoring case, Both strings are identical!\n");
    }
    else{
        printf("Not Equal, ");
        if(strcmp(lower1, lower2) == 0){
            printf("Equal ignoring case, ");
        }
        else{
            printf("Not equal ignoring case, ");
        }

        if(scmp < 0){
            printf("%s comes before %s\n", string1, string2);
        }
        else{
            printf("%s comes before %s\n", string2, string1);
        }
    }
}
